plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -604,7 +605,7 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                         }
 
                         Table icebergTable = catalog.loadTable(session, tableName);
-                        List<ColumnMetadata> columns = getColumnMetadatas(icebergTable.schema());
+                        List<ColumnMetadata> columns = getColumnMetadatas(getCurrentSnapshotSchema(icebergTable));
```

Comment:
Why it's more correct than `icebergTable.schema()`?

With correct code, it looks like the `information_schema.columns` doesn't "see" newly added columns until a new data snapshot is committed.